### PR TITLE
fix(BA-5520): use target ownership policy for VFolder clone max_vfolder_count check

### PR DESCRIPTION
**Pull request by fregataa**
## Summary
- Fix `clone()` and `clone_v2()` methods to check the user's resource policy (not the source project's policy) when validating max_vfolder_count during VFolder cloning
- Previously, cloning a project-owned VFolder incorrectly passed `source_vfolder_data.group` to `get_max_vfolder_count`, which checked the source project's resource policy instead of the requester's user resource policy
- Add regression tests verifying `get_max_vfolder_count` is called with `(user_uuid, None)` when cloning project-owned VFolders

## Test plan
- [x] Unit tests verify `get_max_vfolder_count` is called with correct arguments `(user_uuid, None)` for both `clone()` and `clone_v2()`
- [x] CI passes lint and type checks
- [x] CI passes test suite

Resolves BA-5520

### CODE REVIEW

File: src/ai/backend/manager/services/vfolder/services/vfolder.py [hunk 1]
```diff
@@ -1681,8 +1681,9 @@ async def clone_v2(self, action: CloneVFolderV2Action) -> CloneVFolderV2ActionRe
             user_uuid=action.user_id,
         )
 
+        # clone_v2 always creates user-owned vfolders, so check user resource policy (group_uuid=None)
```
Comment (jopemachine) [inline src/ai/backend/manager/services/vfolder/services/vfolder.py:1684]: If, as the comment suggests, group_uuid must always be passed as None, why is this parameter needed? If it is not actually required, it would be better to remove it.

Comment (seedspirit) [reply src/ai/backend/manager/services/vfolder/services/vfolder.py:1684]: This is because the logic is designed such that when `get_max_vfolder_count` receives a project ID (group_id), it first validates `group_row.resource_policy_row.max_vfolder_count`, and only if the project ID is `None` does it proceed to validate the user’s resource policy.

In other words, since this bug fundamentally stems from the fact that `get_max_vfolder_count` is performing two roles, it would be best to either separate these functions or (since there is no use case for this method to verify the max vfolder count for a group) modify the function to simply retrieve the user’s max vfolder count.

Comment (fregataa) [reply src/ai/backend/manager/services/vfolder/services/vfolder.py:1684]: it is a fix task not a refactoring, I'd like to keep this work as small as possible

Comment (fregataa) [reply src/ai/backend/manager/services/vfolder/services/vfolder.py:1684]: > This is because the logic is designed such that when `get_max_vfolder_count` receives a project ID (group_id), it first validates `group_row.resource_policy_row.max_vfolder_count`, and only if the project ID is `None` does it proceed to validate the user’s resource policy.
> 
> In other words, since this bug fundamentally stems from the fact that `get_max_vfolder_count` is performing two roles, it would be best to either separate these functions or (since there is no use case for this method to verify the max vfolder count for a group) modify the function to simply retrieve the user’s max vfolder count.

get_max_vfolder_count serving dual roles (user vs project policy) is indeed the root cause that made this bug possible. Separating these responsibilities would prevent similar mistakes in the future.

That said, a bug fix PR should stay minimal and focused. Mixing a refactoring into a fix increases review scope and rollback risk. The right approach is to merge this fix as-is and track the refactoring as a follow-up task.
